src/bundle-manifest.ts

```diff
@@ -49,14 +52,12 @@ export class BundleManifest {
    * them, the constructor populates the `bundles` and `files` index properties.
    */
   constructor(bundles: Bundle[], urlMapper: BundleUrlMapper) {
-    const bundleUrls = urlMapper(bundles);
-
-    this.bundles = new Map();
+    this.bundles = urlMapper(Array.from(bundles));
     this.bundleUrlForFile = new Map();
 
-    for (let i = 0; i < bundles.length; ++i) {
-      const bundle = bundles[i], bundleUrl = bundleUrls[i];
-      this.bundles.set(bundleUrl, bundle);
+    for (const bundleMapEntry of this.bundles) {
+      const bundleUrl = bundleMapEntry[0], bundle = bundleMapEntry[1];
+      bundle.url = bundleUrl;
```

Comment:
Seems odd to mutate the inputs to the constructor. Can bundles be created with their URL, or keep this in a map?